sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_pipeline_component_builder.py

```diff
@@ -351,8 +351,6 @@ def _get_name_or_component_name(node: Union[BaseNode, AutoMLJob]):
             # for node name _, treat it as anonymous node with name unset
             if name == "_":
                 continue
-            if name is not None:
```

Comment:
This would be a breaking, any idea to make it not breaking?

Reply:
Updated to lower the node name immediately to avoid breaking change as offline synced.